pkg/query/columnquery.go

```diff
@@ -944,10 +946,37 @@ func getMappingFilesAndLabels(
 		return nil, nil, err
 	}
 
-	labels, err := q.GetProfileMetadataLabels(ctx, []string{}, startTime, endTime)
+	labels, err := q.GetProfileMetadataLabels(ctx, startTime, endTime)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	return mappingFiles, labels, nil
 }
+
+func KWayMerge(arr1, arr2 []string) []string {
```

Comment:
this is not 100% right, we need not a pure k-way merge, but it also needs to de-duplicate

I'd say we should just name the function `merge` and in the code comment say that it does a deduplicating k-way merge, so the two slices that are passed must already be sorted